test/visualTest.ts

```diff
@@ -70,11 +70,107 @@ module powerbi.extensibility.visual.test {
+                let receivedNum = VisualClass.getFromCycledSequence(targetArr, 1, 2);
+                expect(receivedNum).toEqual(5);
+            });
+
+            it("getFromCycledSequence returns array item by exceeded index", () => {
+                let targetArr: number[] = [4, 4, 4, 5, 4, 4];
+                let receivedNum = VisualClass.getFromCycledSequence(targetArr, 9);
+                expect(receivedNum).toEqual(5);
+                receivedNum = VisualClass.getFromCycledSequence(targetArr, 21);
+                expect(receivedNum).toEqual(5);
+            });
+
+            it("getFromCycledSequence returns array item by exceeded index with offset", () => {
+                let targetArr: number[] = [4, 4, 4, 5, 4, 4];
+                let receivedNum = VisualClass.getFromCycledSequence(targetArr, 4, 5);
+                expect(receivedNum).toEqual(5);
+                receivedNum = VisualClass.getFromCycledSequence(targetArr, 14, 7);
+                expect(receivedNum).toEqual(5);
+                receivedNum = VisualClass.getFromCycledSequence(targetArr, 7, 14);
+                expect(receivedNum).toEqual(5);
+            });
+
+            it("getFromCycledSequence returns array item by negative index", () => {
+                let targetArr: number[] = [4, 4, 4, 5, 4, 4];
+                let receivedNum = VisualClass.getFromCycledSequence(targetArr, -2);
+                expect(receivedNum).toBeUndefined();
+            });
+
+            it("getFromCycledSequence returns array item by negative index and positive offset", () => {
+                let targetArr: number[] = [4, 4, 4, 5, 4, 4];
+                let receivedNum = VisualClass.getFromCycledSequence(targetArr, -2, 5);
+                expect(receivedNum).toEqual(5);
+                receivedNum = VisualClass.getFromCycledSequence(targetArr, -2, 1);
+                expect(receivedNum).toBeUndefined();
+            });
+
+            it("getFromCycledSequence returns array item by positive index and negative offset", () => {
+                let targetArr: number[] = [4, 4, 4, 5, 4, 4];
+                let receivedNum = VisualClass.getFromCycledSequence(targetArr, 3, -4);
+                expect(receivedNum).toBeUndefined();
+                receivedNum = VisualClass.getFromCycledSequence(targetArr, 5, -2);
+                expect(receivedNum).toEqual(5);
+            });
+
+            it("getFromCycledSequence returns array item by negative index and negative offset", () => {
+                let targetArr: number[] = [4, 4, 4, 5, 4, 4];
+                let receivedNum = VisualClass.getFromCycledSequence(targetArr, -3, -3);
+                expect(receivedNum).toBeUndefined();
+            });
+        });
+
         describe("DOM tests", () => {
             it("svg element created", () => {
                 expect(visualBuilder.mainElement[0]).toBeInDOM();
             });
 
+            it("no intersections (simplified check)", (done) => {
```

Comment:
Test's name is not clear to me. Let's rename to `words should't intersect each other if words' angle is zero`